<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">
  <!-- Constants -->
  <xacro:property name="base_radius" value="0.05" /> <!-- Radius of the circular base -->
  <xacro:property name="wheel_radius" value="0.03" /> <!-- Radius of the wheels -->
  <xacro:property name="wheel_separation" value="0.12" /> <!-- Distance between the wheels -->
  <xacro:property name="sphere_radius" value="0.15" /> <!-- Radius of the outer sphere -->

  <!-- Materials -->
  <material name="blue">
    <color rgba="0 0 1 1"/>
  </material>
  <material name="black">
    <color rgba="0 0 0 1"/>
  </material>
  <material name="white">
    <color rgba="1 1 1 1"/>
  </material>
  <material name="gray">
    <color rgba="0.5 0.5 0.5 1"/>
  </material>

  <gazebo reference="base_link">
  <material>Gazebo/Blue</material>
  </gazebo>

  <gazebo reference="left_wheel">
    <material>Gazebo/Black</material>
  </gazebo>

  <gazebo reference="right_wheel">
    <material>Gazebo/Black</material>
  </gazebo>

  <gazebo reference="sphere_link">
    <material>Gazebo/White</material>
  </gazebo>
</robot>
